subsys/fs/littlefs_fs.c

```diff
@@ -569,6 +488,50 @@ static int littlefs_statvfs(struct fs_mount_t *mountp,
 	return lfs_to_errno(ret);
 }
 
+/* Backend functions */
+/* Flash map (fmp) backend functions */
+static inline bool lfs_on_fmpdev(struct fs_mount_t *mountp)
+{
+	return !(mountp->flags & FS_MOUNT_FLAG_USE_DISK_ACCESS);
+}
+
+static int lfs_api_read_fmp(const struct lfs_config *c, lfs_block_t block,
+			    lfs_off_t off, void *buffer, lfs_size_t size)
+{
+	const struct flash_area *fa = c->context;
+	size_t offset = block * c->block_size + off;
+
+	int rc = flash_area_read(fa, offset, buffer, size);
+
+	return errno_to_lfs(rc);
+}
+
+static int lfs_api_prog_fmp(const struct lfs_config *c, lfs_block_t block,
+			    lfs_off_t off, const void *buffer, lfs_size_t size)
+{
+	const struct flash_area *fa = c->context;
+	size_t offset = block * c->block_size + off;
+
+	int rc = flash_area_write(fa, offset, buffer, size);
+
+	return errno_to_lfs(rc);
+}
+
+static int lfs_api_erase_fmp(const struct lfs_config *c, lfs_block_t block)
+{
+	const struct flash_area *fa = c->context;
+	size_t offset = block * c->block_size;
+
+	int rc = flash_area_erase(fa, offset, c->block_size);
+
+	return errno_to_lfs(rc);
+}
+
+static int lfs_api_sync_fmp(const struct lfs_config *c)
+{
+	return LFS_ERR_OK;
+}
+
```

Comment:
As you have already introduced the `FS_LITTLEFS_FMP_DEV` - maybe it would be better to move those functions (and also ones for block devices handling) to separate files with proper headers providing declarations ?
Compiling in those files would depend on `FS_LITTLEFS_FMP_DEV` and `FS_LITTLEFS_BLK_DEV`?

Reply:
I don't think this would improve the readability (you would need to search for the routines), it would also have no influence on the generated code. It would also hide the fact that they are not completely separated since we are using the property that if a mount point is not a disk we are assuming that it is flash.

Reply:
Ok.